### PR TITLE
JSON-related changes for enums and required property

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/service/src/Messaging/Models/PurgeMessageQueueResult.cs [hunk 1]
```diff
@@ -23,13 +23,13 @@ protected internal PurgeMessageQueueResult()
         /// <summary>
         /// The Id of the device whose messages are being purged.
         /// </summary>
-        [JsonProperty("deviceId", Required = Required.Always)]
+        [JsonProperty("deviceId")]
```
Comment (drwill-ms) [inline iothub/service/src/Messaging/Models/PurgeMessageQueueResult.cs:26]: We don't need to validate items are in the payload when de/serializing. We can trust the service included it, or if the user didn't include a required item, the service should return an error.

File: iothub/service/src/Registry/Models/AuthenticationMechanism.cs [hunk 2]
```diff
@@ -61,7 +60,6 @@ public X509Thumbprint X509Thumbprint
         /// <summary>
         /// Gets or sets the authentication type.
         /// </summary>
-        [DefaultValue(ClientAuthenticationType.Sas)]
```
Comment (drwill-ms) [inline iothub/service/src/Registry/Models/AuthenticationMechanism.cs:64]: Not sure this even worked, but we don't want SAS to be the default, anyway.

File: provisioning/service/src/Models/CustomAllocationDefinition.cs [hunk 3]
```diff
@@ -13,15 +14,13 @@ public class CustomAllocationDefinition
         /// <summary>
         /// The webhook URL used for allocation requests.
         /// </summary>
-        [JsonProperty("webhookUrl", Required = Required.Always)]
-#pragma warning disable CA1056 // Uri properties should not be strings
-        public string WebhookUrl { get; set; }
-#pragma warning restore CA1056 // Uri properties should not be strings
+        [JsonProperty("webhookUrl")]
+        public Uri WebhookUrl { get; set; }
```
Comment (drwill-ms) [inline provisioning/service/src/Models/CustomAllocationDefinition.cs:18]: Made this a URI and got rid of the suppression.

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

**Review by andyk-ms — APPROVED** [review]

**Review by schoims — APPROVED** [review]

**Pull request merged**